### DISCUSSION TITLE
Computed Field causes failure on model_validate

### DISCUSSION THREAD

**Discussion by bronzeson** [pydantic/pydantic#8580] [Question]
Hi, I would like to know if there is some way around a problem I'm facing 

I have a model 
```py
class Person(BaseModel):
  model_config = ConfigDict(extra="forbid")

  firstName: str
  lastName: str
  
  @computed_field
  def compositeName(self) -> str
    return self.firstName + self.lastName
```

And I model_dump this model and store it.
Later on I model_validate on the JSON that I previously dumped.

I get a failure because extra="forbid" does not allow the computed field "compositeName"

I would like to keep extra="forbid" but allow the computed_field to be an extra field that comes in and is simply discarded and re-computed.

Is this possible or am I doing something fundamentally wrong? Like maybe these should be different types altogether (like Person and PersonSerialized)

**Comment by mfkaroui** [comment]
Sorry for the year + bump but ran across this

extra="ignore" is the right answer here.
forbid is doing exactly as designed raising an error on extra fields

you found a cyclical case where forbid would cause issues.

**Comment by ezsouza** [comment]
Hey @bronzeson! There's a way to keep `extra="forbid"` and still work with computed fields. The issue is that `model_dump()` includes computed fields in the output by default, and when you feed that back to `model_validate()`, pydantic sees them as unknown extra fields.

Two approaches:

**1. Exclude computed fields on dump**

```python
data = person.model_dump(exclude_computed_fields=True)
# Store 'data'...

# Later:
person2 = Person.model_validate(data)  # Works fine now
```

This is probably the cleanest. You only dump the real fields, not the computed ones. When you validate it back, the computed fields get recomputed automatically.

**2. Filter them out before validate**

```python
# If you already dumped with computed fields included
full_data = person.model_dump()

# Remove computed field keys
real_data = {k: v for k, v in full_data.items() 
             if k not in Person.model_computed_fields}

person2 = Person.model_validate(real_data)
```

Either way works, but option 1 is what I'd recommend.

Let me know if it solves your case!
